bindings/c-ffi/src/utils.rs

```diff
@@ -165,11 +165,89 @@ pub(crate) fn generate_keys(bitcoin_network: *const c_char) -> Result<String, Er
     Ok(serde_json::to_string(&res)?)
 }
 
+pub(crate) fn restore_keys(
+    bitcoin_network: *const c_char,
+    mnemonic: *const c_char,
+) -> Result<String, Error> {
+    let bitcoin_network = BitcoinNetwork::from_str(&ptr_to_string(bitcoin_network))?;
+    let mnemonic = ptr_to_string(mnemonic);
+    let res = rgb_lib::restore_keys(bitcoin_network.into(), mnemonic)?;
+    Ok(serde_json::to_string(&res)?)
+}
+
 pub(crate) fn get_address(wallet: &COpaqueStruct) -> Result<String, Error> {
     let wallet = Wallet::from_opaque(wallet)?;
     Ok(wallet.get_address()?)
 }
 
+pub(crate) fn get_btc_balance(
+    wallet: &COpaqueStruct,
+    online: &COpaqueStruct,
+) -> Result<String, Error> {
+    let wallet = Wallet::from_opaque(wallet)?;
+    let online = Online::from_opaque(online)?;
+    let res = wallet.get_btc_balance((*online).clone())?;
+    Ok(serde_json::to_string(&res)?)
+}
+
+pub(crate) fn get_asset_balance(
+    wallet: &COpaqueStruct,
+    asset_id: *const c_char,
+) -> Result<String, Error> {
+    let wallet = Wallet::from_opaque(wallet)?;
+    let asset_id = ptr_to_string(asset_id);
+    let res = wallet.get_asset_balance(asset_id)?;
+    Ok(serde_json::to_string(&res)?)
+}
+
+pub(crate) fn list_btc_transactions(
```

Comment:
```suggestion
pub(crate) fn list_transactions(
```
Please keep the naming consistent. Also fix this on the counterpart method